nion/swift/Inspector.py

```diff
@@ -540,6 +540,23 @@ def _set_property_value(self, value: typing.Optional[typing.Any]) -> None:
 
 
 class DisplayItemPropertyCommandModel(Model.PropertyChangedPropertyModel[typing.Any]):
+    def __init__(self, document_controller: DocumentController.DocumentController,
+                 display_item: DisplayItem.DisplayItem, property_name: str) -> None:
+        super().__init__(display_item, property_name)
+        self.__display_item = display_item
+        self.__document_controller = document_controller
+        self.__property_name = property_name
+
+    def _set_property_value(self, value: typing.Optional[typing.Any]) -> None:
+        command = ChangeDisplayItemPropertyCommand(self.__document_controller.document_model, self.__display_item, self.__property_name, value)
+        command.perform()
+        self.__document_controller.push_undo_command(command)
```

Comment:
I think this needs `if value != self._get_property_value():` to check if the value changed.

First, to describe how I discovered this: editing the caption isn't undo-able with a single Undo without it. It takes two undo's. With this change, it is undo/redo-able.

Putting a `import traceback; traceback.print_stack()` here showed that the value is being set after editing, but then the 'text edited' is received from Qt and then it gets set again. This is a common problem with UI bindings, and my typical workaround is to check if the value changed. I'm not sure whether the equality check should go into `_set_property_value` or done by its caller (in `PropertyChangedPropertyModel`) - let's leave that as an open question for now.

I think something similar needs to go wherever `_set_property_value` is overridden in this file, btw.